widgetssdk/src/main/java/com/glia/widgets/StringProviderImpl.kt

```diff
@@ -1,41 +1,85 @@
 package com.glia.widgets
 
+import android.os.Parcelable
+import androidx.annotation.StringRes
 import com.glia.androidsdk.Glia
 import com.glia.androidsdk.GliaException
-import com.glia.widgets.helper.Logger
 import com.glia.widgets.helper.ResourceProvider
+import kotlinx.parcelize.Parcelize
 
-internal class StringProviderImpl(private val resourceProvider: ResourceProvider) : StringProvider {
+@Parcelize
+data class LocaleString(
+    @StringRes val stringKey: Int,
+    val values: Array<StringKeyPair> = emptyArray()
+): Parcelable {
+
+    companion object {
+        fun from(@StringRes stringKey: Int, vararg values: StringKeyPair = emptyArray()) = LocaleString(stringKey, arrayOf(*values))
```

Comment:
My first attempt was also to use the secondary contractor but for some reason during compilation it return an error 
![image](https://github.com/salemove/android-sdk-widgets/assets/10000880/8f1e51eb-1827-4e84-b99b-298a4897a42a)

But after your other suggestions I have tried it with list instead of array and it seems to work in this case with secondary constractor.

Reply:
Wow! Indeed, We have `arrays` in both cases, I think if you change `array` to a `list` it might be possible with `values.toList`.